src/Sema.zig

```diff
@@ -25263,7 +25263,13 @@ fn zirBuiltinExtern(
     if (!ty.isPtrAtRuntime(mod)) {
         return sema.fail(block, ty_src, "expected (optional) pointer", .{});
     }
-    if (!try sema.validateExternType(ty.childType(mod), .other)) {
+
+    const child_ty = ty.childType(mod);
+    const is_function_child_ty = child_ty.zigTypeTag(mod) == .Fn;
+    if (is_function_child_ty and !ty.isConstPtr(mod)) {
+        return sema.fail(block, ty_src, "pointer to extern function should be 'const'", .{});
```

Comment:
I'm not sure if this check should even exist but if it should then it should be moved to `validateExternType` so that it applies everywhere and `should` should be replaced with `must`.

Reply:
done